pisek/jobs/jobs.py

```diff
@@ -83,7 +83,9 @@ def __init__(self, name: str) -> None:
         self.dirty = False  # Prints something to console?
 
         self.prerequisites = 0
-        self.required_by: list[tuple["PipelineItem", Optional[str]]] = []
+        self.required_by: list[
+            tuple["PipelineItem", Optional[str], Callable[[Any], bool]]
+        ] = []
```

Comment:
Nit: It might be beneficial to use a named tuple, since it isn't obvious what each element means.